### PR TITLE
Feature/nested parameters

### PR DESCRIPTION
**Pull request by Kibnet**
I added support for nested parameters via JSON. Changed the rules for forming the base path. This is all to facilitate the migration of the appsettings.json in the Vault.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,13 +47,32 @@ Alternatively, you can configure Vault connection using next environmnt variable
 ## Preparing secrets in Vault
 
 You need to store your secrets with special naming rules.
-First of all, all secrets should use KV2 storage and have prefix `data/{app_alias}/`.
-For example, if your app has alias `sampleapp` and you want to have configuration option `ConnectionString` your secret path would be `data/sampleapp/ConnectionString`.
+First of all, all secrets should use KV2 storage and have prefix `{app_alias}/{env}`.
```
Comment (MrZoidberg) [inline README.md:50]: Have you checked this with real Vault data? I remember there was a problem with reading and writing with VaultSharp secrets that are not prefixed with `data`. I'm totally ok with removing this requirement, but I'd like to be confident we don't break things.

File: README.md [hunk 2]
```diff
@@ -47,13 +47,32 @@ Alternatively, you can configure Vault connection using next environmnt variable
...
+For example, if your app has alias `sampleapp` and environment `producton` and you want to have configuration option `ConnectionString` your secret path would be `sampleapp/producton`.
```
Comment (MrZoidberg) [inline README.md:51]: What do you think if we reverse the path to be `{env}/{app_alias}`? That would simplify creating Vault policies by env name.

**Comment by MrZoidberg** [comment]
@Kibnet Hey, great work, thanks! I have a few comments, please take a look. Also, the integration test is failing.

File: README.md (hunk 1 above)
Comment (Kibnet) [reply README.md:50]: Yes, I checked on real data. Works. I recommend you try it too.

File: README.md (hunk 2 above)
Comment (Kibnet) [reply README.md:51]: This description is purely advisory in nature, and everyone can do whatever they want. If this is important to you, you can reflect this point in the Readme.

**Comment by Kibnet** [comment]
> @Kibnet Hey, great work, thanks! I have a few comments, please take a look. Also, the integration test is failing.

As for the tests, I couldn't run them on the local machine. So I tested it manually. Mikhail, can you help me fix the tests?

**Comment by MrZoidberg** [comment]
@Kibnet sure, will do that. I think you couldn't run the test because they require docker to be installed. But I will help with that.

**Comment by MrZoidberg** [comment]
@Kibnet I've fixed test here https://github.com/MrZoidberg/VaultSharp.Extensions.Configuration/commit/0044cf48d727f2ca506e216422ba59e6ff4b867a.

**Comment by MrZoidberg** [comment]
Anyway I'll check this version with my services if these changes are ok

**Comment by Kibnet** [comment]
> Anyway I'll check this version with my services if these changes are ok

You will have to make changes there, because the storage structure has changed.

**Comment by MrZoidberg** [comment]
@Kibnet yep, I see and that may be a problem.

**Pull request merged**